q2_moshpit/plugin_setup.py

```diff
@@ -381,6 +383,45 @@
                 'analyses.'
 )
 
+plugin.methods.register_function(
+    function=q2_moshpit.eggnog.build_diamond_db,
+    inputs={
+        'sequences': FeatureData[ProteinSequence],
+    },
+    outputs=[('diamond_db', ReferenceDB[Diamond])],
+    parameters={
+        "threads": Int % Range(1, None),
+    },
+    input_descriptions={
+        'sequences': "Artifact containing protein reference database file "
+                     "in FASTA format."
+    },
+    output_descriptions={
+        'diamond_db': "Artifact containing a binary DIAMOND database file."
+    },
+    parameter_descriptions={
+        "threads": "Number of CPU threads. By default, the program will "
+                   "auto-detect and use all available virtual cores on the "
+                   "machine.",  # Expose
+        # "verbose": "verbose console output",  # Dont? Simply default no?
+        # "log": "enable debug log",  # Dont?
+        # "quiet": "disable console output",  # Dont? Already managed by qiime
+        # "taxonmap": "protein accession to taxid mapping file",  # Expose?
+        # "taxonnodes": "taxonomy nodes.dmp from NCBI",  # Expose?
+        # "taxonnames": "taxonomy names.dmp from NCBI",  # Expose?
+        # "file-buffer-size": "file buffer size in bytes (default=67108864)",
+        #                     # Expose?
+        # "no-unlink": "Do not unlink temporary files.",  # Expose?
+        # "ignore-warnings": "Ignore warnings",  # Expose?
+        # "no-parse-seqids": "Print raw seqids without parsing"  # Expose?
```

Comment:
Maybe we discuss later which of these options we would like to expose the user to. I think `taxonmap`, `taxonnodes`, and `taxonnames` are of special importance judging by [the documentation](https://github.com/bbuchfink/diamond/wiki/3.-Command-line-options#:~:text=DIAMOND%20database%20file.-,%2D%2Dtaxonmap%20%3Cfile%3E,-Path%20to%20mapping), since it seems that they offer some extended functionality.

Reply:
- I added `taxon<>` files as optional inputs in the last commits. 
- Related PR and issues regarding these optional inputs:
   - bokulich-lab/q2-types-genomics#66
   - bokulich-lab/q2-types-genomics#67